streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -88,6 +92,32 @@ public void process(K key, Change<V> change) {
 
     }
 
+    private class MaterializedKTableFilterProcessor extends AbstractProcessor<K, Change<V>> {
```

Comment:
You could probably merge this class and `KTableFilterProcessor` into a single class. if the storeName is null you don't bother getting it from the context. And then in `process` if the store is null don't do the `put`

Alternatively you could extract the common code into a method that can be used by both impls